leakpro/reporting/utils.py

```diff
@@ -28,30 +25,21 @@ def prepare_priavcy_risk_report(
         NotImplementedError: Check if the report for the privacy game is implemented.
 
     """
-    audit_report.REPORT_FILES_DIR = "privacy_meter/report_files"
     if save_path is None:
-        save_path = log_dir
-
-    if configs["privacy_game"] in [
-        "privacy_loss_model",
-        "avg_privacy_loss_training_algo",
-    ]:
-        # Generate privacy risk report for auditing the model
-        if len(audit_results) == 1 and configs["privacy_game"] == "privacy_loss_model":
-            ROCCurveReport.generate_report(
-                metric_result=audit_results[0],
-                save=True,
-                filename=f"{save_path}/ROC.png",
-                configs=configs,
-            )
-            SignalHistogramReport.generate_report(
-                metric_result=audit_results[0],
-                save=True,
-                filename=f"{save_path}/Histogram.png",
-            )
-        else:
-            raise ValueError(
-                f"{len(audit_results)} results are not enough for {configs['privacy_game']})"
-            )
-    else:
-        raise NotImplementedError(f"{configs['privacy_game']} is not implemented yet")
+        raise ValueError("Please provide a save path for the report")
+
+    if audit_results is None:
+        raise ValueError("Please provide the audit results")
+
+    # Generate privacy risk report for auditing the model
+    ROCCurveReport.generate_report(
+        metric_result=audit_results,
+        save=True,
+        filename=f"{save_path}/ROC.png",
+        configs=configs,
+    )
+    SignalHistogramReport.generate_report(
```

Comment:
Adding a report generator function to abstract_mia and calling whatever type of report we want from each attack itslef.

Reply:
I think this should be its own issue. I created it as issue#35